src/engine/Filter.cpp

```diff
@@ -388,9 +389,13 @@ void Filter::computeResultFixedValue(ResultTable* result) const {
         case SparqlFilter::LANG_MATCHES:
           getEngine().filter(*static_cast<vector<RT>*>(subRes->_fixedSizeData),
                              [this, &l](const RT& e) {
-                               return ad_utility::endsWith(
-                                   getIndex().idToString(e[l]),
-                                   this->_rhsString);
+                               std::optional<string> entity =
+                                   getIndex().idToOptionalString(e[l]);
+                               if (!entity) {
+                                 return true;
+                               }
+                               return ad_utility::endsWith(entity.value(),
+                                                           this->_rhsString);
```

Comment:
How about writing a Function, that returns this language filtering lambda? This is used multiple times again in the code below. We could e.g. write
`getEngine().filter(*static_cast<vector<RT>*>(subRes->_fixedSizeData), getLanguageFilterLambda(l)`
where `getLanguageFilterLambda` is a member function of the filter class.
Additionally, is the captured variable `l` not a numeric type and thus can/should be captured by value?

Reply:
(This was not really possible in c++11 since returning a "lambda" without the overhead of a function was a pain, but now (since c++14) we can write
```
auto getLambda(int toCapture) {
  return [this, toCapture](const int x) {return this._memberVal + x < toCapture; };
}
```

Reply:
Good point, let me investigate. I was hesitant to make the changes too big but this is already somewhat intrusive so it might actually be a good time to clean this up.

Reply:
I actually found a way (pushed) to get rid of multiple lambdas as well as a lot of the repetition